docs/api-reference/geo-layers/tile-layer.md

```diff
@@ -261,6 +261,34 @@ Receives arguments:
 
 - `tile` (Object) - the tile that has been cleared from cache.
 
+
+##### `onViewportChange` (Function, optional)
+
+`onViewportChange` is a function that is called when the viewport changes or all tiles in the current viewport have been loaded. A function (`getRenderedFeatures`) to calculate the features rendered in the current viewport is passed as a JSON to this callback function.
+
+`getRenderedFeatures` only works if pickable property is true.
+
+It is not recommended to call `getRenderedFeatures` every time `onViewportChange` is executed, instead, better use `getRenderedFeatures` inside a debounce function.
```

Comment:
Based on your implementation, I do not think `getRenderedFeatures` works if called immediately. `onViewportChange` is triggered before `renderLayers`, therefore will not render the correct sub layers to the picking buffer.